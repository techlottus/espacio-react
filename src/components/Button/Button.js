

import React, { useEffect } from 'react'


export const Button = React.memo(({ data, onClick }) => {



  const buttonRef = React.createRef();
  
  useEffect(() => {
    buttonRef.current.data = {
      id: data.id || null,
      type: data.type  || 'primary',
      title: data.title || '',
      size: data.size || 'small',
      icon: data.icon || '',
      lyIcon: data.lyIcon,
      disabled: data.disabled,
      isExpand: data.isExpand,
      test: data.test || ''
    }
  }, [data]) // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    buttonRef.current.addEventListener('onClick', onClick);
  }, []) // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-button ref={buttonRef}></lottus-button>
    </>
  )
})


