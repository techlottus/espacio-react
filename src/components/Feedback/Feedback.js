import React, { useEffect } from 'react';
import "./Feedback.scss";

export const Feedback = ({data, onRight, text, html=() => {return}, isNoti=false}) => {

    const feedbackRef = React.createRef();
  
    useEffect(() => {
      feedbackRef.current.data = {
        left: data.left || {},
        right: data.right || {},
        type: data.type || '',
        isTextEvent: data.isTextEvent,
        textEvent: data.textEvent || '',
      };
      
    },[data,text]) // eslint-disable-line react-hooks/exhaustive-deps

    useEffect(() => {
      feedbackRef.current.addEventListener('onRight', onRight)
    },[]) // eslint-disable-line react-hooks/exhaustive-deps
    
    return (
      <>
        <lottus-feedback ref={feedbackRef}> 
          <div content="" className={`feedback-content ${isNoti ? "feedbackcontentmessagetext" : ""}`}>{text || html()}</div>
        </lottus-feedback>
      </>
    )
  }
