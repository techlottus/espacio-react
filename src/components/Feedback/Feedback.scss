@import "../../App.scss";

.feedback-content {
  font-family: "Nunito", sans-serif;
  line-height: pixeltorem(19px);
  font-size: pixeltorem(14px);

  @media only screen and (min-width: map-get($grid-breakpoints, m)) {
    font-size: pixeltorem(16px);
    line-height: pixeltorem(22px);
  }
}

.feedbackcontentmessagetext {
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 2; /* number of lines to show */
  line-clamp: 2;
  -webkit-box-orient: vertical;
}