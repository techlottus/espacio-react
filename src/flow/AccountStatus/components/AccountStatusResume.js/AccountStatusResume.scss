@import '../../../../App.scss';

.accountstatusresume-container {
  font-family: "Nunito", sans-serif;
  width: calc(100% - #{pixeltorem(32px)});
  position: relative;
  display: flex;
  flex-direction: column;
  background-color: map-get($palette-neutral, n00);
  padding: pixeltorem(16px) pixeltorem(16px) 0 pixeltorem(16px);

  @media only screen and (min-width: map-get($grid-breakpoints, m)) {
    border-radius: pixeltorem(8px);
    padding: pixeltorem(36px) pixeltorem(36px) 0 pixeltorem(36px); 
  }

  .accountstatusresumetitle {
    font-size: pixeltorem(18px);
    line-height: pixeltorem(24px);
    margin-top: pixeltorem(36px);
    margin-bottom: pixeltorem(36px);
    font-family: "Nunito Bold", sans-serif;
  }

  > .accountstatusresumebtn {
    margin-top: pixeltorem(36px);
    margin-bottom: pixeltorem(64px);
    display: flex;
    flex-direction: column;
    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      flex-direction: row;
    }
    .downloadbtn {
      margin-bottom: pixeltorem(16px);
      @media only screen and (min-width: map-get($grid-breakpoints, m)) {
        margin-right: pixeltorem(16px);
      }
    }
  }

  .accountstatuserror {
    text-align: center;
    font-size: pixeltorem(14px);
    font-family: "Nunito Bold", sans-serif;
    padding: pixeltorem(16px) 0;
  }

}
.termsfeedback {
  @extend .alertfeedbackcmp;
}