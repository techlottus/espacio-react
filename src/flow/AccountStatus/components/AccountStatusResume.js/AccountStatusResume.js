import React, { useEffect, useState } from "react";
import { CardProfile } from "../../../../components/Card/CardProfile";
import {
  cardProfile,
  checkInvoiceBtn,
  downloadBtn,
} from "../../../../constants/AccountStatus.constant";
import "./AccountStatusResume.scss";
import { Button } from "../../../../components/Button/Button";
import AccountStatusTable from "../AccountStatusTable/AccountStatusTable";
import { useDispatch, useSelector } from "react-redux";
import { capitalize, formatString } from "../../../../helpers/formatString";
import { getAccountReportService } from "../../../../middlewares/accountStatusMiddleware";
import { typesRequestErrors } from "../../../../constants/error.constant";
import ScreenEmpty from "../../../Screens/ScreenAccountStatus/ScreenEmptyAccountStatus";
import { notiObs } from "../../../../observables/notificationObs";
import { typesNoti } from "../../../../types/typeNoti";
import { useHistory } from "react-router";
import { sendInfoTM } from "../../../../tagging/services/sendInfoTagManager";
import { buttonDownloadAccountStatusTag } from "../../../../tagging/flows/accountStatusTag";
import { getImageOfAssets } from "../../../../helpers/getImages";

const AccountStatusResume = () => {
  const history = useHistory();

  const { texts } = useSelector((state) => state.texts);
  const [downloadButton] = useState({
    ...downloadBtn,
    isExpand: window.innerWidth < 991,
  });

  const [checkButton] = useState({
    ...checkInvoiceBtn,
    isExpand: window.innerWidth < 991,
  });

  const [next, setNext] = useState(false);

  const [profile, setProfile] = useState({
    ...cardProfile,
    image: getImageOfAssets(texts?.accountStatus?.images?.user)
  });

  const accountStatus = useSelector((state) => state.accountStatus);

  const dispatch = useDispatch();

  useEffect(() => {
    if (accountStatus.profile !== null)
      setProfile((state) => {
        return {
          ...state,
          name: capitalize(accountStatus.profile.fullName),
          items: [
            formatString(accountStatus.profile.enrollmentNumber),
            formatString(accountStatus.profile.institute),
            formatString(accountStatus.profile.academicLevel),
            formatString(accountStatus.profile.program),
          ],
        };
      });
  }, [accountStatus.profile]);

  useEffect(() => {
    if (
      accountStatus.errors &&
      accountStatus.errors[typesRequestErrors.getAccountStatus].isError
    ) {
      notiObs.next({
        type: typesNoti.error,
        text: accountStatus.errors[typesRequestErrors.getAccountStatus].msg,
      });
    }
  }, [accountStatus.errors]);

  useEffect(() => {
    if (next) {
      history.push("/account-status-invoices-query");
      setNext(false);
    }
  }, [next]);

  const hanldeResume = () => {
    return accountStatus.debitPeriods !== null ? (
      <>
        {accountStatus.debitPeriods.length === 0 ? (
          <ScreenEmpty />
        ) : (
          <div>
            <div className="accountstatusresumetitle">
              {texts?.accountStatus?.resumeTitle}
            </div>
            <div className="accountstatusresumetable">
              <AccountStatusTable />
            </div>
          </div>
        )}
      </>
    ) : (
      ""
    );
  };

  const handleError = () => {
    return (
      <div className="accountstatuserror">
        {texts?.accountStatus?.errorMessage}
      </div>
    );
  };

  return (
    <>
      <div className="accountstatusresume-container">
        {accountStatus.profile === null || accountStatus.debitPeriods === null
          ? handleError()
          : ""}
        <div className="accountstatusresumeperfil">
          {accountStatus.profile !== null ? <CardProfile data={profile} /> : ""}
        </div>
        {hanldeResume()}
        <div className="accountstatusresumebtn">
          <div className="downloadbtn">
            <Button
              data={downloadButton}
              onClick={() => {
                dispatch(getAccountReportService());
                sendInfoTM(window, buttonDownloadAccountStatusTag);
              }}
            />
          </div>
          <Button
            data={checkButton}
            onClick={() => {
              setNext(true);
            }}
          />
        </div>
      </div>
    </>
  );
};

export default AccountStatusResume;
