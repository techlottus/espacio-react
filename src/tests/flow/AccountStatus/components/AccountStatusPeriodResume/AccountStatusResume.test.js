import React from "react";
import { mount } from "enzyme";
import AccountStatusResume from "../../../../../flow/AccountStatus/components/AccountStatusResume.js/AccountStatusResume";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { typesRequestErrors } from "../../../../../constants/error.constant";
import { Feedback } from "../../../../../components/Feedback/Feedback";
import { act } from "react-dom/test-utils";
import { fireEvent } from "@testing-library/dom";
import { Button } from "../../../../../components/Button/Button";
import { BrowserRouter } from "react-router-dom";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

window.scrollTo = jest.fn();

let initState = {
  accountStatus: {
    profile: null,
    movements: null,
    errors: {
      [typesRequestErrors.getAccountStatus]: {
        isError: false,
        msg: null,
      },
      [typesRequestErrors.getAccountMovements]: {
        isError: false,
        msg: null,
      },
    },
    programs: null,
    success: {
      isSuccess: false,
      msg: null,
    },
    debitPeriods: null,
  },
  texts: {
    accountStatus: {
      errorMessage: "hola",
      resumeTitle: "hola",
    },
  },
};

const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <BrowserRouter>
        <AccountStatusResume history={historyMock} />
      </BrowserRouter>
    </Provider>
  );
};

describe("Test <AccountStatusResume/>", () => {
  beforeEach(() => {
    store = mockStore(initState);
  });

  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
  });

  test("should show error", () => {
    initState = {
      ...initState,
      accountStatus: {
        ...initState.accountStatus,
        errors: {
          ...initState.accountStatus.errors,
          [typesRequestErrors.getAccountStatus]: {
            isError: true,
            msg: "Error",
          },
        },
      },
    };
    const wrapper = handleMount(initState);
  });

  describe("should handleResume", () => {
    test("ToBeEmpty", () => {
      initState = {
        ...initState,
        accountStatus: {
          ...initState.accountStatus,
          debitPeriods: [],
        },
      };
      const wrapper = handleMount(initState);
      expect(wrapper.find("ScreenEmpty").exists()).toBe(true);
    });
    test("ToBeFull", () => {
      initState = {
        ...initState,
        accountStatus: {
          ...initState.accountStatus,
          debitPeriods: [
            {
              periodCode: "202146",
              periodDescription: "DGETI  F21-M21",
              startDate: "2021-01-25",
              endDate: "2021-05-17",
              debtAmount: 5960.0,
            },
            {
              periodCode: "202147",
              periodDescription: "DGETI  J21-A21",
              startDate: "2021-05-24",
              endDate: "2021-09-08",
              debtAmount: 100.0,
            },
          ],
        },
      };
      const wrapper = handleMount(initState);
      expect(wrapper.find("AccountStatusTable").exists()).toBe(true);
    });
    test("ToBeNull", () => {
      initState = {
        ...initState,
        accountStatus: {
          ...initState.accountStatus,
          debitPeriods: null,
        },
      };
      const wrapper = handleMount(initState);
      expect(wrapper).toMatchSnapshot();
    });
  });

  test("close feedback delete error", () => {
    initState = {
      ...initState,
      accountStatus: {
        ...initState.accountStatus,
        errors: {
          [typesRequestErrors.getAccountStatus]: {
            isError: true,
            msg: "error en cuenta",
          },
        },
      },
    };
    const wrapper = handleMount(initState);
  });

  test("should download account state", () => {
    initState = {
      ...initState,
      accountStatus: {
        ...initState.accountStatus,
        errors: {
          [typesRequestErrors.getAccountStatus]: {
            isError: true,
            msg: "error en cuenta",
          },
        },
      },
    };
    const wrapper = handleMount(initState);

    act(() => {
      wrapper.find(".downloadbtn").find(Button).props().onClick();
    });
    wrapper.update();
  });

  test("should push ", () => {
    initState = {
      ...initState,
      accountStatus: {
        ...initState.accountStatus,
      },
    };
    const wrapper = handleMount(initState);
    act(() => {
      wrapper
        .find(".accountstatusresumebtn")
        .find(Button)
        .last()
        .props()
        .onClick();
    });
    wrapper.update();
    console.log(historyMock.push.mock);
    //expect(historyMock.push.mock.calls[0][0]).toEqual('/account-status-invoices-query');
  });

  test("should", () => {
    initState = {
      ...initState,
      accountStatus: {
        ...initState.accountStatus,
        profile: {
          fullName: "Jorge Hernandez",
          enrollmentNumber: "200029578",
          institute: "UTC DISTANCIA",
          academicLevel: "LICENCIATURA 334",
          program: "ADMINISTRACIÃ“N DE EMPRESAS",
        },
      },
    };
    const wrapper = handleMount(initState);
  });

  test("should", () => {
    initState = {
      ...initState,
      accountStatus: {
        ...initState.accountStatus,
        debitPeriods: []
      },
    };
    const wrapper = handleMount(initState);
  });
});
