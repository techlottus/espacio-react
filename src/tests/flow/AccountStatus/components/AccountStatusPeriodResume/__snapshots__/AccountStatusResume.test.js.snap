// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AccountStatusResume/> should handleResume ToBeNull 1`] = `
<div
  className="accountstatusresume-container"
>
  <div
    className="accountstatuserror"
  />
  <div
    className="accountstatusresumeperfil"
  />
  <div
    className="accountstatusresumebtn"
  >
    <div
      className="downloadbtn"
    >
      <lottus-button />
    </div>
    <lottus-button />
  </div>
</div>
`;

exports[`Test <AccountStatusResume/> should init component 1`] = `
<div
  className="accountstatusresume-container"
>
  <div
    className="accountstatuserror"
  />
  <div
    className="accountstatusresumeperfil"
  />
  <div
    className="accountstatusresumebtn"
  >
    <div
      className="downloadbtn"
    >
      <lottus-button />
    </div>
    <lottus-button />
  </div>
</div>
`;
